Library/MapleBacon/MapleBacon/MapleBaconConfig.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  MapleBaconConfig.swift
+//  MapleBacon
+//
+//  Created by Danilo Topalovic on 10.03.16.
+//  Copyright © 2016 Zalando SE. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+/**
+ * MapleBacon Config
+ */
+public class MapleBaconConfig {
+    
+    /// singleton param
+    public static let sharedConfig: MapleBaconConfig = MapleBaconConfig()
+    
+    /// Holds the storage part of the config
+    var storage: Storage
+    
+    /**
+     Private Constructor
+     inits default config
+     */
+    private init() {
+        self.storage = MapleBaconConfig.Storage.defaultStorage()
+    }
+    
+    /**
+     * Storage Class
+     * Holds all configuration for the storage
+     * part of MapleBacon
+     */
+    class Storage {
+        
+        /// used for idents and/or paths of the storage engines
+        var defaultStorageName: String = ""
+        /// used for dispatch and/or serial queues
+        var queueLabel: String = ""
+        /// basepath of the storage (disk)
+        var baseStoragePath: String = ""
+        /// the rootUUIDNamespace should be either ns:URL or ns:DNS -> http://www.ietf.org/rfc/rfc4122.txt
+        var rootUUIDNamespace: String = ""
+        /// should be the bundle-id or other unique app ident
+        var baseUUIDName: String = ""
+        /// use UUID Idents instead of clean sha1
```

Comment:
I won't comment every single comment :trollface: . But this one is particularly harmful: I can follow along that "Ident" is short for identifier (though this isn't a common abbreviation in English) but then this talks about UUID and a "clean" sha1. Why is one better than the other – are clean sha1's better? I don't know. This should go to some length to at least explain possible tradeoffs or why it's important to choose one or the other.